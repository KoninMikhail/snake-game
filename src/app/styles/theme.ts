import {colors, ThemeColors} from './theme_parts/colors';
import {gaps, ThemeGaps} from './theme_parts/gaps';
import {icons, ThemeIcons} from './theme_parts/icons';
import {ThemeTypography, typography} from './theme_parts/typography';
import {animations, ThemeAnimations} from '@styles/theme_parts/animations';

/**
 * @typedef IBaseTheme
 */
export interface IBaseTheme {
    animations: ThemeAnimations;
    colors: ThemeColors;
    gaps: ThemeGaps;
    icons: ThemeIcons;
    typography: ThemeTypography;
}

/**
 * Theme
 */
export const theme: IBaseTheme = {
    colors,
    animations,
    gaps,
    icons,
    typography,
};
