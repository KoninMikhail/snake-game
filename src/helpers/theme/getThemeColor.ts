import { ThemeColors } from '@styles/theme_parts/colors';
import { IBaseTheme } from '@styles/theme';

/**
 * @name getThemeColor
 *
 * Get the color from theme.
 *
 * @param theme - The theme object
 * @param color - The color you want to get from the theme.
 * @returns A function that takes a theme and a color and returns the color from the theme.
 */
const getThemeColor: GetFromThemeFn<IBaseTheme, ThemeColors> = (theme, color) => {
    return theme.colors[color];
};

export default getThemeColor;
