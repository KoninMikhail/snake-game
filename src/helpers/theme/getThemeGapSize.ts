import {IBaseTheme} from '@styles/theme';
import {ThemeGaps} from '@styles/theme_parts/gaps';

/**
 * @name getThemeGapSize
 *
 * Get the gap size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the gap you want to get.
 * @returns A function that takes a theme and a size and returns the theme.gaps[size]
 */
const getThemeGapSize: GetFromThemeFn<IBaseTheme, ThemeGaps> = (theme, size) => theme.gaps[size];

export default getThemeGapSize;
