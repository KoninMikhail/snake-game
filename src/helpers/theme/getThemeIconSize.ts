import { IBaseTheme } from '@styles/theme';
import { ThemeIcons } from '@styles/theme_parts/icons';

/**
 * @name getThemeIconSize
 *
 * Get the icon size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the icon you want to get.
 * @returns A function that takes a theme and a size and returns the theme.icons[size]
 */
const getThemeIconSize: GetFromThemeFn<IBaseTheme, ThemeIcons> = (theme, size) => {
    return theme.icons[size];
};

export default getThemeIconSize;
