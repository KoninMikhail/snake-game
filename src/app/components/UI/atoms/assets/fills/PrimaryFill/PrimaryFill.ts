import { useTheme } from 'styled-components';
import getThemeColor from '@/helpers/theme/getThemeColor';
import { cloneElement } from 'react';
import { IFillProps } from '@ui/atoms/assets/fills/Fill.interface';

/**
 * @name PrimaryFill
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use it for fill text and icons to primary theme color.
 *
 * @param  children - element for pass styles
 * @returns {JSX.Element}
 */
export const PrimaryFill = ({ children }: IFillProps) => {
    const theme = useTheme();
    const outputColor = getThemeColor(theme, 'primary');
    return cloneElement(children, {
        style: {
            color: outputColor,
        },
    });
};
