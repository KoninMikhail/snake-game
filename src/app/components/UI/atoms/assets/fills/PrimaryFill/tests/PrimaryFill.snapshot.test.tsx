import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { PrimaryFill } from '@ui/atoms/assets/fills/PrimaryFill/PrimaryFill';
import { IBaseTheme } from '@styles/theme';

/**
 * Mocks
 */
const fakeTheme = {
    colors: {
        primary: 'deeppink',
    },
} as unknown as IBaseTheme;

/**
 * Tests
 */
describe('Primary IFillProps', () => {
    const testClassName = 'testElement';
    const testId = 'testElementID';

    it('equal snapshots', () => {
        const { container } = render(
            <ThemeProvider theme={fakeTheme}>
                <PrimaryFill>
                    <div data-testid={testId} className={testClassName}>
                        Test Element
                    </div>
                </PrimaryFill>
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
});
