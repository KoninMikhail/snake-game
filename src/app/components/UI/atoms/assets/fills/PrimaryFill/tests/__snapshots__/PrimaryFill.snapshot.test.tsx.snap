// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary IFillProps equal snapshots 1`] = `
<div>
  <div
    class="testElement"
    data-testid="testElementID"
    style="color: deeppink;"
  >
    Test Element
  </div>
</div>
`;
